docs/rfcs/002-branch-management.md

```diff
@@ -0,0 +1,95 @@
+# Branch Management From Command Palette
+
+## Status
+
+Proposed
+
+## Summary
+
+Efficient branch management that is in line with existing Atom conventions.
+
+## Motivation
+
+Current branch management is incomplete (e.g. no deleting of branches) and not consistent with other conventions found in the Atom UI for comparable actions. It is hard to switch or create new branches with just keyboard input.
+
+## Explanation
+
+The new branch selector exists largely in the [command palette](https://github.com/atom/command-palette). The only UI element is the pre-existing `branch-view` tile in the status bar:
+
+![existing branch view tile](https://cldup.com/CE_61W1uBk.png)
+
+### Command Palette
+
+The command palette contains the following new commands:
+- Create new branch (`github:create-new-branch`)
+- Checkout branch (`github:checkout-branch`)
+- Delete branch (`github:delete-branch`)
+- Rename branch (`github:rename-branch`)
```

Comment:
I second @simurai's suggestion about collapsing these into a single command, or at least checkout and create. That'd be consistent with the way that github.com does it:

<img width="179" alt="screen shot 2018-04-24 at 9 51 58 pm" src="https://user-images.githubusercontent.com/17565/39222269-cda9ee0e-4809-11e8-9467-ecf18ebc39a1.png">

I think it'd be nice to not have to lose your search and place in the list to do a bunch of branch-related tasks at once.